src/lib/logical_query_plan/sort_node.cpp

```diff
@@ -9,14 +9,23 @@
 
 namespace opossum {
 
-OrderByDefinition::OrderByDefinition(const ColumnID column_id, const OrderByMode order_by_mode)
-    : column_id(column_id), order_by_mode(order_by_mode) {}
+OrderByDefinition::OrderByDefinition(const LQPColumnOrigin& column_origin, const OrderByMode order_by_mode)
+    : column_origin(column_origin), order_by_mode(order_by_mode) {}
 
-SortNode::SortNode(const std::vector<OrderByDefinition>& order_by_definitions)
+SortNode::SortNode(const OrderByDefinitions& order_by_definitions)
     : AbstractLQPNode(LQPNodeType::Sort), _order_by_definitions(order_by_definitions) {}
 
-std::shared_ptr<AbstractLQPNode> SortNode::_deep_copy_impl() const {
-  return std::make_shared<SortNode>(_order_by_definitions);
+std::shared_ptr<AbstractLQPNode> SortNode::_deep_copy_impl(const std::shared_ptr<AbstractLQPNode>& left_child,
+                                                           const std::shared_ptr<AbstractLQPNode>& right_child) const {
+  OrderByDefinitions order_by_definitions;
+  std::transform(_order_by_definitions.begin(), _order_by_definitions.end(), std::back_inserter(order_by_definitions),
+                 [&](const auto& order_by_definition) {
+                   return OrderByDefinition{
+                       this->left_child()->deep_copy_column_origin(order_by_definition.column_origin, left_child),
+                       order_by_definition.order_by_mode};
+                 });
```

Comment:
```
for(const auto &definition : _order_by_definitions) {
  order_by_definitions.emplace_back(this->left_child()->deep_copy_column_origin(order_by_definition.column_origin, left_child),
                                    order_by_definition.order_by_mode);
}
```

is shorter by more than 100 characters and is easier to read. Any reason to use transform and back_inserter here? Same in other places.

Reply:
I agree, but then everyone tells me to "express my intent" and use stuff like std::generate and transorm and fill and whatnot....